packages/web3-utils/src/hash.ts

```diff
@@ -159,8 +159,12 @@ const getType = (arg: Sha3Input): [string, EncodingTypes] => {
 		type = 't' in arg ? arg.t : arg.type;
 		value = 'v' in arg ? arg.v : arg.value;
 
-		// otherwise try to guess the type
-	} else {
+		type = type.toLowerCase() === 'bigint' ? 'int' : type;
```

Comment:
Looks like from muhammads' [example  ](https://github.com/web3/web3.js/issues/6327) number strings also need to be converted to int

Reply:
I [think](https://github.com/web3/web3.js/issues/6327#issuecomment-1667521008) these are diff data types and 4.x is aligned with solidity , as in solidity sha3 output is different for string vs number.
```
contract ShaHashTest {
	    function func10() external pure returns (bytes32) {
        		return keccak256(abi.encodePacked(int(10))) ;}

        function funcStr10() external pure returns (bytes32) {
        		return keccak256(abi.encodePacked(string("10"))) ;}
}
```
for `int` 10 its : 0xc65a7bb8d6351c1cf70c95a316cc6a92839c986682d98bc35f958f4883f9d2a8
and for `bytes/string` '10' its : 0x1a192fabce13988b84994d4296e6cdc418d55e2f1d7f942188d4040b94fc57ac